tune_sklearn/tune_basesearch.py

```diff
@@ -414,15 +415,19 @@ def _fit(self, X, y=None, groups=None, **fit_params):
             best_config = analysis.get_best_config(
                 metric=metric, mode="max", scope="last")
             self.best_params = self._clean_config_dict(best_config)
-            if not check_partial_fit(
-                    self.estimator) and check_warm_start_ensemble(
-                        self.estimator):
+
+            self.best_estimator_ = clone(self.estimator)
+            if self.early_stop_type == EarlyStopping.warm_start_ensemble:
                 logger.info("tune-sklearn uses `n_estimators` to warm "
                             "start, so this parameter can't be "
                             "set when warm start early stopping. "
                             "`n_estimators` defaults to `max_iters`.")
-                self.best_params["n_estimators"] = self.max_iters
-            self.best_estimator_ = clone(self.estimator)
+                if check_is_pipeline(self.estimator):
```

Comment:
@Yard1 I think this was accidentally left out in the early stopping pipeline pr. Could you just confirm that this is correct?

For ensemble early stopping, because we add 1 to `n_estimators` and then call fit for each call to _trainable to incrementally train ensembles, the final `n_estimators` is the same as the number of calls to _trainable (which should be `max_iters` for the best set of hyperparameters). However, you can't set this directly for pipelines, so I'm trying to set it to the base estimator here. Is it okay to index in like this and set `n_estimators`? Or do I need to add it to the `self.best_params` dict using the underscore syntax (e.g `"classify__min_samples_split": [2, 3, 4]`) and set it for the overall pipeline.

Reply:
No, it is ok to do it like this. You can modify each pipeline step separately without issues. I opted to use the underscore syntax due to preference, but either works.